### PR TITLE
doc: add upgrade page

### PR DESCRIPTION
**Pull request by victorjulien**

### CODE REVIEW

File: doc/userguide/upgrade.rst [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+Configuration Updates
+~~~~~~~~~~~~~~~~~~~~~
+
+New versions of Suricata will occationally include updated config files:
+``classification.config`` and ``reference.config``. Since the Suricata
+installation will not overwrite these if they exist, they should be manually
+updated. If there are no local modifications they can simply be overwritten
+by the ones Suricata supplies.
+
```
Comment (norg) [inline doc/userguide/upgrade.rst]: It's also relevant to some parts of the suricata.yaml as well
Should we also add a diff which settings changed there or just recommend people to use diff and read the changelog/new config?

File: doc/userguide/upgrade.rst [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+Major updates include new features, new default settings and often also
+remove features.
```
Comment (norg) [inline doc/userguide/upgrade.rst]: Link to deprecation policy?

File: doc/userguide/upgrade.rst [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+
+Upgrading 4.1 to 5.0
+--------------------
+
+Major changes
+~~~~~~~~~~~~~
+- New protocols enabled by default: snmp (new config only)
+- New protocols disabled by default: rdp, sip
+- New defaults: nfs, smb, tftp, krb5 ntp are all enabled by default
```
Comment (norg) [inline doc/userguide/upgrade.rst]: New protocols defaults to make it more clear it's about the same topic

File: doc/userguide/upgrade.rst (hunk 1 above)
Comment (victorjulien) [reply doc/userguide/upgrade.rst]: The diff is quite large and a bit too technical I think. I would prefer a more textual overview of changes.

**Comment by victorjulien** [comment]
Merged with some improvements through #4288

**Pull request closed without merging**